open-company/open-company-web
# Experimental section switch

Trello: https://trello.com/c/7nBALFyT/2247-experimental-section-switching-perf-improvement

To test:
- with user A
- open AP
- asap switch to Follow-ups section
- [x] do you see follow-ups post appear?
- now switch to section X
- now switch to section Y
- now expand post 1
- with user B: add a comment to another post 2 mentioning user A
- with user A click on the notification to nav to the post
- with user A now user the browser back arrow
- [x] does the history reflect your real path?
- [x] check also the browser forward button
- now you should be on post 2 again
- [x] does the button says Go back to All post (not back to section)?
- click it
- [x] are you on all posts?
- now nav to section M using the navigation bar
- nav to section N using the navigation bar
- nav to section O using the navigation bar
- [x] check again the browser back and forth buttons
- go forward in browser history to go again to section O
- expand another post
- with user B comment mentioning user A on post 3
- with user A click on the internal notification to open it
- with user B comment mentioning user A on post 4
- with user A click on the internal notification to open it
- with user B comment mentioning user A on post 5
- with user A click on the internal notification to open it
- [x] with user A check that the browser history is still correct
- do any other test on history and section switching that you can think of

Drafts test:
- open the Network tab of dev tools
- open AP
- click Draft
- [x] do you see the draft posts appear and the ghost screen going away?
- now type the drafts url and press Enter to nav directly to it
- [x] did it load the drafts posts?
- click on section A in navigation bar
- [x] check from Network tab that section A has been reloaded after you navigated into it
- click on section B in navigation bar
- [x] check from Network tab that section B has been reloaded after you navigated into it
- click on Drafts again
- [x] check from network tab that the drafts board has been reloaded on the last nav to it
